libraries/rodeos/include/eosio/key_value.hpp

```diff
@@ -403,6 +403,10 @@ class kv_table {
          return compare(b) >= 0;
       }
 
+      explicit operator bool() {
+         return compare(this) != 0;
```

Comment:
An iterator is always equal to itself.  Also, there is no matching overload of compare.